tests/kafka.rs

```diff
@@ -0,0 +1,80 @@
+use futures::{StreamExt, TryStreamExt};
+use rdkafka::{
+    config::ClientConfig,
+    consumer::{stream_consumer::StreamConsumer, Consumer},
+    producer::{FutureProducer, FutureRecord},
+    Message,
+};
+use std::time::Duration;
+
+use testcontainers::{clients, core::RunArgs, images::kafka};
+
+#[tokio::test]
+async fn test_produce_and_consume_messages() {
+    let docker = clients::Cli::default();
+    let kafka_node = docker.run_with_args(
+        kafka::Kafka::default(),
+        RunArgs::default().with_mapped_port((kafka::KAFKA_PORT, kafka::KAFKA_PORT)),
+    );
+
+    let bootstrap_servers = format!("localhost:{}", kafka_node.get_host_port(kafka::KAFKA_PORT));
+
+    let producer: FutureProducer = ClientConfig::new()
+        .set("bootstrap.servers", &bootstrap_servers)
+        .set("message.timeout.ms", "5000")
+        .create()
+        .expect("Failed to create Kafka FutureProducer");
+
+    let consumer: StreamConsumer = ClientConfig::new()
+        .set("group.id", "testcontainer-rs")
+        .set("bootstrap.servers", &bootstrap_servers)
+        .set("session.timeout.ms", "6000")
+        .set("enable.auto.commit", "false")
+        .set("auto.offset.reset", "earliest")
+        .create()
+        .expect("Failed to create Kafka StreamConsumer");
+
+    let topic = "test-topic";
+    consumer
+        .subscribe(&[&topic])
+        .expect("Failed to subscribe to a topic");
+
+    let number_of_messages_to_produce = 5_usize;
+    let expected: Vec<String> = (0..number_of_messages_to_produce)
+        .map(|i| format!("Message {}", i))
+        .collect();
```

Comment:
Personally, I usually prefer to use turbo-fish syntax on the actual function that requires the type parameters. The reason is that type-hints easily get stale if you change the expression whereas usage of turbo-fish syntax gives the compiler just enough information to continue with type inference by itself. That in turn makes future code changes easier because type inference automatically adapts to your new code.

```suggestion
    let expected = (0..number_of_messages_to_produce)
        .map(|i| format!("Message {}", i))
        .collect::<Vec<_>();
```

Reply:
Makes sense. Converted all the type annotations to turbo-fish in 81be19a.